sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -82,8 +187,55 @@ export class TableCheckpointStore implements CheckpointStore {
    *  - `tracingOptions`: Options for configuring tracing.
    * @returns A promise that resolves when the checkpoint has been updated.
    */
-  async updateCheckpoint(/* checkpoint: Checkpoint */): Promise<void> {
-    console.log("no checkpoint to update");
-    throw new Error("not implemented");
+  async updateCheckpoint(checkpoint: Checkpoint) {
+    let PARTITIONKEY =
+      checkpoint.eventHubName +
+      " " +
+      checkpoint.fullyQualifiedNamespace +
+      " " +
+      checkpoint.consumerGroup +
+      " " +
+      "Checkpoint";
+
+    const entity1: customCheckpoint = {
+      partitionKey: PARTITIONKEY,
+      rowKey: checkpoint.partitionId,
+      consumerGroup: checkpoint.consumerGroup,
+      fullyQualifiedNamespace: checkpoint.fullyQualifiedNamespace,
+      eventHubName: checkpoint.eventHubName,
+      sequenceNumber: checkpoint.sequenceNumber,
+      offset: checkpoint.offset,
+      partitionId: checkpoint.partitionId
+    };
+
+    let entitiesIter = this._tableClient.listEntities<Checkpoint>({
+      queryOptions: { filter: odata`PartitionKey eq ${PARTITIONKEY}` }
+    });
+    let i = 0;
+    for await (const ent of entitiesIter) {
+      ent.offset;
+      i++;
+    }
+
+    if (i > 0) {
+      let checkpoints: Checkpoint[] = [];
+      checkpoints = await this.listCheckpoints(
+        checkpoint.fullyQualifiedNamespace,
+        checkpoint.eventHubName,
+        checkpoint.consumerGroup
+      );
+      for (const checkpnt of checkpoints) {
+        if (checkpnt.partitionId == checkpoint.partitionId) {
+          checkpnt.offset = checkpnt.offset;
+          checkpnt.sequenceNumber = checkpnt.sequenceNumber;
+        } else {
+          await this._tableClient.upsertEntity(entity1);
+        }
+      }
+    } else {
+      await this._tableClient.upsertEntity(entity1);
+    }
```

Comment:
It seems like you should just be able to upset the entity without doing these checks. What benefit are we gaining with these checks?